contributors/design-proposals/storage/raw-block-ro.md

```diff
@@ -0,0 +1,139 @@
+# Raw Block Read-Only Consumption in Kubernetes
+
+Authors: erinboyd@, screeley44@, mtanino@
+
+This document presents a proposal for honoring the read-only setting for raw block devices in the CRI into Kubernetes. This document builds on the design
+outlined in #805 & #1265
+
+# Terminology
+* Raw Block Device - a physically attached device devoid of a filesystem
+* Raw Block Volume - a logical abstraction of the raw block device as defined by a path
+* Filesystem on Block - a formatted (ie xfs) filesystem on top of a raw block device
+
+# Goals
+* Create consistent interaction with container runtime for readOnly block devices 
+
+  
+# Specific use case of storage consumption for read-only:
+  * An admin wishes to expose a block volume for read-only consumption to an application without write permissions. Example
+ applications might be a database with data to be used for analytics processing in which the results of processing are written to another fileystem or application.  
+
+ 
+# Design Overview
+
+ Today in the container runtime, the raw block devices are defaulted to 'rwm' which provides, "read", "write" and "mknod". There exists such use cases as
+outlined above in which we want to restrict the user to only access the block device as a reader. This support already exists in the container runtime and should
+be supported in Kube. 
+In addition, in the container spec today we use the 'readOnly' parameter for the same purpose when using a filesystem. The proposal is to add this same parameter to the volumeDevices definition in the API.
```

Comment:
yeah, make mention that in the existing container spec for VolumeMounts object (which is what VolumeDevices was generally based on) it has a ```readOnly bool```

Reply:
Given volume devices are only backed by pvc/pv objects, I sort of expected the readonly aspect to be controlled by the accessmode on the pvc or the readOnly field on the PV. Which field governs if the PV and Pod disagree?

Reply:
@liggitt, I don't see a readOnly field in the PV?

Reply:
@liggitt today the accessMode is essentially a label used for binding and it's not enforced. Thus, configured correctly, the admin 'should' set the accessMode to RO in this case to indicate a user wanting to use it that is how the PV is configured. However...we don't enforce this today, thus having the user specify the read-only we can then change the behavior of the runtime to honor this. Does this make sense?

Reply:
you're right... it's on the PVCVolumeSource. Given that volumeDevices are only used with pvcVolumeSources, it seems like we already have a spot to indicate this in the API